### ISSUE TITLE
In-cluster Docker build fails with: error parsing HTTP 405 response body: invalid character 'M' 

### ISSUE THREAD

**Issue by eysi09** [garden-io/garden#1996]
### Current Behavior

In-cluster building fails with:

```
Failed building frontend. Here is the output:
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
Command "/home/solomon/.garden/tools/kubectl/804c71b3e6ad5976/kubectl --context=gke --kubeconfig=/home/solomon/gardenio/garden-platform/services/infra/gke-cluster-dev-f29377c0.yaml 
--namespace=garden-system exec -i garden-docker-daemon-546769bf5b-nsx9x -c docker-daemon -- /bin/sh -c docker push 127.0.0.1:5000/garden-platform/frontend:v-d37f411cf2" failed with code 1:
error parsing HTTP 405 response body: invalid character 'M' looking for beginning of value: "Method not allowed
"
command terminated with exit code 1
Here's the full output:
The push refers to repository [127.0.0.1:5000/garden-platform/frontend]
error parsing HTTP 405 response body: invalid character 'M' looking for beginning of value: "Method not allowed
```

after running the cleanup cluster command (`garden plugins kubernetes cleanup-cluster-registry --env=<env-name>`).

This happens because the Docker registry is stuck in RO mode. 

### Expected behavior

Garden should ensure the cleanup command doesn't leave the Docker registry in a RO state.

### Reproducible example

Try running `garden plugins kubernetes cleanup-cluster-registry --env=<env-name>` and then `garden build`.

### Workaround

Run the cleanup command again: `garden plugins kubernetes cleanup-cluster-registry --env=<env-name>`.

If that doesn't work, the following should:

1. Remove the docker registry with: `helm uninstall garden-docker-registry -n garden-system`
2. Run Cluster init with: `garden plugins kubernetes cluster-init --env <env-name>`

### Suggested solution(s)

N/A

### Additional context

I believe this is related to #1831. At least this error happens frequently (or always?) after the issue described in #1831 happens.

**Comment by eysi09** [comment]
Update: It appears that this can also be resolved by simply running `garden plugins kubernetes cleanup-cluster-registry --env=<env-name` which is a far simpler workaround. I'll update the initial issue to reflect that.

**Comment by edvald** [comment]
Closed by #2042
